src/modules/xkeyboard.cpp

```diff
@@ -289,26 +289,19 @@ namespace modules {
       }
 
       const string& layout = vec[0];
+
       if (layout.empty()) {
         m_log.warn("%s: layout-icon '%s' is invalid: there must always be a layout defined", name(), it);
         continue;
       }
 
-      string variant;
-      string icon;
-
-      if (size == 2) {
-        variant = layouticonset::VARIANT_ANY;
-        icon = vec[1];
-      } else {
-        variant = vec[1];
-        icon = vec[2];
-      }
+      const string& variant = size == 2 ? layouticonset::VARIANT_ANY : vec[1];
```

Comment:
I think there might be a problem with this solution because VARIANT_ANY is a `static constexpr const char* ` and the compiler will need to create a string from it to assign this const reference but this string may be created on the stack and the reference may not live long enough to be used correctly once the ternary operator finishes.
I may be wrong as I'm not fluent in c++ const_expr

Reply:
I don't think that is an issue here. The way I understand it is that from `VARIANT_ANY` a string object is created, a reference to which is stored in `variant`. The lifetime of this object is prolonged to match the lifetime of the reference.